### DISCUSSION TITLE
Whether it is possible to skip certain workers in a mosec workflow

### DISCUSSION THREAD

**Discussion by tyz1994** [mosecorg/mosec#349] [Q&A] [answered]
Hello everyone,

I am wondering if workers must be executed in a sequential order or if it is possible to skip certain workers in the workflow. 
For example, we have three workers, [preprocess, inference, postprocess]. Inference worker takes the most time-consuming work. If we find some errors in preprocess, we want to skip inference worker and go directly to postprocess, rather than queueing util inference worker is free. 

Can anyone provide insight or resources on this topic? Thank you.

**Comment by kemingy** [comment, marked as answer]
You can raise mosec exceptions so it will skip the following steps and return to the client directly.

**Reply by tyz1994** [reply]
Great idea. It works. I'm now using `raise ValidationError("xxx")` to send text response for tasks that require early quit.
